atomix/cluster/src/main/java/io/atomix/raft/storage/system/MetaStore.java

```diff
@@ -160,6 +163,27 @@ public synchronized MemberId loadVote() {
     return id.isEmpty() ? null : MemberId.from(id);
   }
 
+  public synchronized long loadFlushedIndex() {
+    try {
+      metaFileChannel.read(metaBuffer, 0);
+    } catch (final IOException e) {
+      throw new StorageException(e);
+    }
+    return serializer.readFlushedIndex(new UnsafeBuffer(metaBuffer), VERSION_LENGTH);
+  }
+
+  public synchronized void storeFlushedIndex(final long index) {
+    log.trace("Store last flushed index {}", index);
+
+    try {
+      serializer.writeFlushedIndex(index, new UnsafeBuffer(metaBuffer), VERSION_LENGTH);
+      metaFileChannel.write(metaBuffer, 0);
+      metaFileChannel.force(true);
```

Comment:
Since this is now in critical path of execution, it would be better to open the file channel with `StandardOpenOption.DSYNC`. Then no need to call `force()`. It is supposed to be better performant as it requires one less syscall.